net/tcp/tcp_callback.c

```diff
@@ -258,12 +258,15 @@ uint16_t tcp_datahandler(FAR struct net_driver_s *dev,
       iob_concat(conn->readahead, iob);
     }
 
-#ifdef CONFIG_NET_TCP_RECV_CONTIG
+#ifdef CONFIG_NET_TCP_RECV_PACK
   /* Merge an iob chain into a continuous space, thereby reducing iob
    * consumption.
    */
 
-  conn->readahead = iob_contig(conn->readahead);
+  if (conn->readahead->io_flink != NULL)
```

Comment:
why need check

Reply:
If it is not chain and does not need pack, reduce copy once